qbraid/runtime/batch.py

```
@@ -0,0 +1,414 @@
+        return self.open()
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:
+        """Close the batch session.
+
+        Delegates to :meth:`close` for context-var reset and backend
+        close, then runs the ``on_all_complete`` callback if registered.
+        Errors from ``close()`` and the callback are logged, not raised.
+        """
+        try:
+            self.close()
+        except Exception as err:  # pylint: disable=broad-exception-caught
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)